db/rdbms/src/main/java/io/camunda/db/rdbms/write/service/CorrelatedMessageWriter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.db.rdbms.write.service;
+
+import io.camunda.db.rdbms.sql.CorrelatedMessageMapper;
+import io.camunda.db.rdbms.sql.HistoryCleanupMapper.CleanupHistoryDto;
+import io.camunda.db.rdbms.sql.ProcessBasedHistoryCleanupMapper;
+import io.camunda.db.rdbms.write.domain.CorrelatedMessageDbModel;
+import io.camunda.db.rdbms.write.queue.ContextType;
+import io.camunda.db.rdbms.write.queue.ExecutionQueue;
+import io.camunda.db.rdbms.write.queue.QueueItem;
+import io.camunda.db.rdbms.write.queue.WriteStatementType;
+import java.time.OffsetDateTime;
+
+public class CorrelatedMessageWriter {
+
+  private final ExecutionQueue executionQueue;
+  private final CorrelatedMessageMapper mapper;
+
+  public CorrelatedMessageWriter(
+      final ExecutionQueue executionQueue, final CorrelatedMessageMapper mapper) {
+    this.executionQueue = executionQueue;
+    this.mapper = mapper;
+  }
+
+  public void create(final CorrelatedMessageDbModel correlatedMessage) {
+    executionQueue.executeInQueue(
+        new QueueItem(
+            ContextType.CORRELATED_MESSAGE,
+            WriteStatementType.INSERT,
+            // Using a composite key made from subscription and message keys
+            correlatedMessage.subscriptionKey() + "_" + correlatedMessage.messageKey(),
```

Comment:
🔧 I think in the camunda exporter this is the other way around, `MESSAGE_KEY_SUBSCRIPTION_KEY`. Not a big deal but we should keep them aligned.

💭 This make me wonder if we might need a common util class to calculate this key